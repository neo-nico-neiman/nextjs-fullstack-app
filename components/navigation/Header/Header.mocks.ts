import { IHeader } from './Header';

const base: IHeader = { 
    // add props
};

const alt: IHeader = {
    // add props
};

export const mockHeaderProps = {
    base,
    alt,
};