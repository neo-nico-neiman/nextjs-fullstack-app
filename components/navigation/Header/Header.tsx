import AuthButton from '@components/buttons/AuthButton';
import Link from 'next/link';

export interface IHeader extends React.ComponentPropsWithoutRef<'header'> {}

const Header: React.FC<IHeader> = ({ className, ...headerProps }) => {
  return (
    <header
      {...headerProps}
      className={`w-full flex flex-row justify-between items-center ${className}`}
    >
      <div className="space-x-5 m-4">
        <Link href="/">
          <a className="hover:underline">About</a>
        </Link>
        <Link href="/">
          <a className="hover:underline">Store</a>
        </Link>
      </div>

      <div className="space-x-5 m-4">
        <Link href="/">
          <a className="hover:underline hidden sm:inline">Gmail</a>
        </Link>
        <Link href="/">
          <a className="hover:underline hidden sm:inline">Images</a>
        </Link>
        <AuthButton />
      </div>
    </header>
  );
};

export default Header;
