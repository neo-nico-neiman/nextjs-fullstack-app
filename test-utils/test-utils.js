// test-utils.js
import Providers from '@state/providers';
import { render } from '@testing-library/react';

const customRender = (ui, options = {}) =>
  render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from '@testing-library/react';
// override render method
export { customRender as render };
