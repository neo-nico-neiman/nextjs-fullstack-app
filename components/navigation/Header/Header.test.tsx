import { render, screen } from '@test-utils/test-utils';
import '@testing-library/jest-dom';

import Header from './Header';

describe('Header', () => {
  test('Should match the snapshot', () => {
    const { asFragment } = render(<Header />);
    expect(asFragment).toMatchSnapshot();
  });

  test('Should contain links', () => {
    render(<Header />);
    expect(screen.getByRole('link', { name: /about/i })).toBeInTheDocument();
    expect(screen.getByRole('link', { name: /store/i })).toBeInTheDocument();
    expect(screen.getByRole('link', { name: /gmail/i })).toBeInTheDocument();
    expect(screen.getByRole('link', { name: /images/i })).toBeInTheDocument();
    expect(
      screen.getByRole('button', { name: /sign in/i })
    ).toBeInTheDocument();
  });
});
