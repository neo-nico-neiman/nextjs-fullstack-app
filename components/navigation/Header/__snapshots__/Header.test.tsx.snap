// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Should match the snapshot 1`] = `[Function]`;
